subsys/bluetooth/host/id.c

```diff
@@ -361,14 +361,20 @@ int bt_id_set_adv_private_addr(struct bt_le_ext_adv *adv)
 		return 0;
 	}
 
-	err = bt_rpa_create(bt_dev.irk[adv->id], &rpa);
-	if (!err) {
-		err = bt_id_set_adv_random_addr(adv, &rpa);
-		if (!err) {
-			atomic_set_bit(adv->flags, BT_ADV_RPA_VALID);
+	if (!atomic_test_bit(&bt_dev.rpa_valid[adv->id], BT_ADV_RPA_VALID)) {
```

Comment:
With my proposed change this becomes `atomic_test_bit(bt_dev.rpa_valid, adv->id)`

Reply:
Removed the flag